.github/workflows/nightly/process-result.ts

```diff
@@ -0,0 +1,192 @@
+          timeReport[matrix.project].minEnv = `${matrix.os_name}, ${matrix.package_manager}`;
+        }
+      } else {
+        timeReport[matrix.project] = {
+          min: matrix.duration,
+          max: matrix.duration,
+          minEnv: `${matrix.os_name}, ${matrix.package_manager}`,
+          maxEnv: `${matrix.os_name}, ${matrix.package_manager}`,
+        };
+      }
+    }
+  });
+
+  let resultPkg = `
+    \`\`\`
+    | Project                        | Time                      |
+    |--------------------------------|---------------------------|`;
+
+  function mapProjectTime(proj: string, section: 'min' | 'max'): string {
+    return `${humanizeDuration(timeReport[proj][section])} (${timeReport[proj][`${section}Env`]})`;
+  }
+
+  function durationIcon(proj: string, section: 'min' | 'max'): string {
+    const duration = timeReport[proj][section];
+    if (duration < 12 * 60) return `${section} ✅`;
+    if (duration < 15 * 60) return `${section} ❗`;
+    return `${section} ❌`;
+  }
+
+  Object.keys(timeReport).forEach(proj => {
+    resultPkg += `\n| ${proj.padEnd(30)} |                           |`;
+    resultPkg += `\n| ${durationIcon(proj, 'min').padStart(29)} | ${mapProjectTime(proj, 'min').padEnd(25)} |`;
+    resultPkg += `\n| ${durationIcon(proj, 'max').padStart(29)} | ${mapProjectTime(proj, 'max').padEnd(25)} |`;
+  });
+  resultPkg += `\`\`\``;
+
+  let resultPm = `
+    \`\`\`
+    | PM   | Total time  |
+    |------|-------------|`;
+  Object.keys(pmReport).forEach(pm => {
+    resultPm += `\n| ${pm.padEnd(4)} | ${humanizeDuration(pmReport[pm as keyof typeof pmReport]).padEnd(11)} |`;
+  });
+  resultPm += `\`\`\``;
+
+  return {
+    codeowners: Array.from(codeowners).join(','),
+    slack_message: trimSpace(result),
+    slack_proj_duration: trimSpace(resultPkg),
+    slack_pm_duration: trimSpace(resultPm),
+    has_golden_failures: hasGoldenFailures.toString(),
+  };
+}
+
+function setOutput(key: string, value: string) {
+  const outputPath = process.env.GITHUB_OUTPUT;
+  if (!outputPath) {
+    console.warn(`GITHUB_OUTPUT not set. Skipping output for "${key}".`);
+    return;
+  }
+
+  if (value.includes('\n')) {
+    const delimiter = `EOF_${key}_${Date.now()}`;
+    fs.appendFileSync(outputPath, `${key}<<${delimiter}\n${value}\n${delimiter}\n`);
+  } else {
+    fs.appendFileSync(outputPath, `${key}=${value}\n`);
+  }
+}
+
+// Main execution
+if (require.main === module) {
+  try {
+    const combinedInput = process.argv[2]
+      ? process.argv[2]
+      : fs.readFileSync(0, 'utf-8').trim();
+
+    const combined: MatrixResult[] = JSON.parse(combinedInput);
+    const results = processResults(combined);
+
+    Object.entries(results).forEach(([key, value]) => {
+      setOutput(key, value);
+    });
+  } catch (error) {
+    console.error('Error processing results:', error);
+    process.exit(1);
+  }
+}
+
+export { processResults, MatrixResult, ProcessedResults };
```

Comment:
I don't think these are imported anywhere? I would remove them and then we can remove the `require.main` check since it's assumed this is a script to be run, and there are no exports.